Core/Lib/BusinessDocumentGenerator.php

```diff
@@ -27,33 +27,99 @@
  */
 class BusinessDocumentGenerator
 {
+    /**
+     * New document generated.
+     *
+     * @var mixed
+     */
+    private $newDoc;
```

Comment:
No. Why newdoc as a class property?

Reply:
Because:
- Is needed to obtain the document to recalculate totals of this new document when we finish to add new lines to the same document. On clone the total is the same, but when we add new lines the total changes.
- Is needed to give a message with a link with new document generated to this document.